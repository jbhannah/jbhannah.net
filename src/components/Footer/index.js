import React from 'react'
import PropTypes from 'prop-types'
import Link from 'gatsby-link'

const year = (new Date()).getFullYear()

const Footer = ({ title, className }) => (
  <footer className={className}>
    <div>
      Copyright Â© {year}
      {' '}<Link to="/">{title}</Link>.
      {' '}<a href="https://creativecommons.org/license/by-sa/4.0/" target="_blank" rel="license">Some Rights Reserved</a>.
      {' '}<a href="https://github.com/jbhannah/jbhannah.net" target="_blank">Source on GitHub</a>.
    </div>
  </footer>
)

Footer.propTypes = {
  title: PropTypes.string,
}

export default Footer
