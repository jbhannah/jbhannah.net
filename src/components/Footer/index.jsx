import PropTypes from "prop-types"
import React from "react"
import Link from "../Link"

const year = new Date().getFullYear()

const Footer = ({ title, className }) => (
  <footer className={className}>
    <div>
      Copyright Â© {year} <Link href="/">{title}</Link>.{" "}
      <Link
        href="https://creativecommons.org/licenses/by-sa/4.0/"
        target="_blank"
        rel="license"
      >
        Some Rights Reserved
      </Link>
      .{" "}
      <Link href="https://github.com/jbhannah/jbhannah.net" target="_blank">
        Source on GitHub
      </Link>
      .
    </div>
  </footer>
)

Footer.propTypes = {
  title: PropTypes.string,
}

export default Footer
