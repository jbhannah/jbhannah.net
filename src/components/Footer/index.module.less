@import "../../assets/css/_mixins";
@import "../../assets/css/_variables";

.footer {
  &:extend(.content-width-column);

  align-self: flex-end;
  font-size: 0.75rem;
  margin: 1rem auto;

  @media @mq[lg] {
    margin-left: @header-width;
  }

  @media @mq[xl] {
    margin-left: 0;
  }
}
