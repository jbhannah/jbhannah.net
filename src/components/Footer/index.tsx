import styled from "@emotion/styled"
import * as React from "react"
import { contentWidthColumn, headerWidth, mq } from "../../utils/styles"
import { Link } from "../Link"

interface FooterProps {
  fullname: string | null
  pronouns: string | null
}

const year = new Date().getFullYear()

const _Footer = styled.footer({
  ...contentWidthColumn,
  alignSelf: "flex-end",
  fontSize: "0.75rem",
  margin: "1rem auto",
  [mq.lg]: { marginLeft: headerWidth },
  [mq.xl]: { marginLeft: 0 },
})

const Footer: React.FunctionComponent<FooterProps> = ({
  fullname,
  pronouns,
}) => (
  <_Footer>
    <div>
      Copyright Â© {year} <Link href="/">{fullname}</Link> ({pronouns}).{" "}
      <Link
        href="https://creativecommons.org/licenses/by-sa/4.0/"
        rel="license"
      >
        Some Rights Reserved
      </Link>
      . Emoji by <Link href="https://twemoji.twitter.com/">Twemoji</Link>.
      <Link href="https://github.com/jbhannah/jbhannah.net">
        Source on GitHub
      </Link>
      .
    </div>
  </_Footer>
)

export default Footer
