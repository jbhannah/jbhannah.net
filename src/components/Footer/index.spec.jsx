import { shallow } from "enzyme"
import MockDate from "mockdate"
import React from "react"
import Footer from "."

MockDate.set("2019-02-05T01:17:07.551Z")

const TestFooter = (props) => shallow(<Footer {...props} />)

describe("Footer", () => {
  const tree = TestFooter({ title: "Test Site" })

  it("renderes correctly", () => {
    expect(tree).toMatchSnapshot()
  })
})
