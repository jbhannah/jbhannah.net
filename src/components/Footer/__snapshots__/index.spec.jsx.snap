// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renderes correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    title="Test Site"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Copyright © 
        2019
         
        <Link
          href="/"
        >
          Test Site
        </Link>
        .
         
        <Link
          href="https://creativecommons.org/licenses/by-sa/4.0/"
          rel="license"
        >
          Some Rights Reserved
        </Link>
        .
         
        <Link
          href="https://github.com/jbhannah/jbhannah.net"
        >
          Source on GitHub
        </Link>
        .
      </div>,
      "css": Array [
        Object {
          "margin": "0 auto",
          "maxWidth": "42rem",
          "padding": "0 1rem",
          "width": "100%",
        },
        Object {
          "@media only screen and (min-width: 59rem)": Object {
            "marginLeft": "12rem",
          },
          "@media only screen and (min-width: 70rem)": Object {
            "marginLeft": 0,
          },
          "alignSelf": "flex-end",
          "fontSize": "0.75rem",
          "margin": "1rem auto",
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Copyright © ",
          2019,
          " ",
          <Link
            href="/"
          >
            Test Site
          </Link>,
          ".",
          " ",
          <Link
            href="https://creativecommons.org/licenses/by-sa/4.0/"
            rel="license"
          >
            Some Rights Reserved
          </Link>,
          ".",
          " ",
          <Link
            href="https://github.com/jbhannah/jbhannah.net"
          >
            Source on GitHub
          </Link>,
          ".",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Copyright © ",
        2019,
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Test Site",
            "href": "/",
          },
          "ref": null,
          "rendered": "Test Site",
          "type": [Function],
        },
        ".",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Some Rights Reserved",
            "href": "https://creativecommons.org/licenses/by-sa/4.0/",
            "rel": "license",
          },
          "ref": null,
          "rendered": "Some Rights Reserved",
          "type": [Function],
        },
        ".",
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Source on GitHub",
            "href": "https://github.com/jbhannah/jbhannah.net",
          },
          "ref": null,
          "rendered": "Source on GitHub",
          "type": [Function],
        },
        ".",
      ],
      "type": "div",
    },
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Copyright © 
          2019
           
          <Link
            href="/"
          >
            Test Site
          </Link>
          .
           
          <Link
            href="https://creativecommons.org/licenses/by-sa/4.0/"
            rel="license"
          >
            Some Rights Reserved
          </Link>
          .
           
          <Link
            href="https://github.com/jbhannah/jbhannah.net"
          >
            Source on GitHub
          </Link>
          .
        </div>,
        "css": Array [
          Object {
            "margin": "0 auto",
            "maxWidth": "42rem",
            "padding": "0 1rem",
            "width": "100%",
          },
          Object {
            "@media only screen and (min-width: 59rem)": Object {
              "marginLeft": "12rem",
            },
            "@media only screen and (min-width: 70rem)": Object {
              "marginLeft": 0,
            },
            "alignSelf": "flex-end",
            "fontSize": "0.75rem",
            "margin": "1rem auto",
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Copyright © ",
            2019,
            " ",
            <Link
              href="/"
            >
              Test Site
            </Link>,
            ".",
            " ",
            <Link
              href="https://creativecommons.org/licenses/by-sa/4.0/"
              rel="license"
            >
              Some Rights Reserved
            </Link>,
            ".",
            " ",
            <Link
              href="https://github.com/jbhannah/jbhannah.net"
            >
              Source on GitHub
            </Link>,
            ".",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Copyright © ",
          2019,
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Test Site",
              "href": "/",
            },
            "ref": null,
            "rendered": "Test Site",
            "type": [Function],
          },
          ".",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Some Rights Reserved",
              "href": "https://creativecommons.org/licenses/by-sa/4.0/",
              "rel": "license",
            },
            "ref": null,
            "rendered": "Some Rights Reserved",
            "type": [Function],
          },
          ".",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Source on GitHub",
              "href": "https://github.com/jbhannah/jbhannah.net",
            },
            "ref": null,
            "rendered": "Source on GitHub",
            "type": [Function],
          },
          ".",
        ],
        "type": "div",
      },
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
